### ISSUE TITLE
Apollo Federation Example

### ISSUE THREAD

**Issue by jacob-lester** [Urigo/graphql-modules#558]
The [example code](https://github.com/Urigo/graphql-modules/tree/master/examples/apollo-federation) for Apollo Federation integration is throwing this error: 

```
GraphQL-Modules Error: Schema is not valid!
   - #Module #C:\Users\...\src\modules\app.module.ts doesn't have a valid schema!
   -- Unknown directive "key".

   Possible solutions:
    - Check syntax errors in typeDefs
    - Make sure you import correct dependencies
```

https://github.com/Urigo/graphql-modules/issues/496 says it was fixed in 0.7.6 however I can't get the code to run (also tried 0.7.7 as well).

**Comment by ardatan** [comment]
It works for me. Do you use exactly same project or modified one?

**Comment by jacob-lester** [comment]
> Do you use exactly same project or modified one?

Using the example project code directly.

**Comment by ardatan** [comment]
I wasn't sure because of the error. As you can see there is no `modules/app.module.ts` in Apollo Federation example.

**Comment by jacob-lester** [comment]
Oh, sorry about that. I copied the error message from my own implementation discussed in another issue. Anyway, the error is still the same.

**Comment by ardatan** [comment]
Could you share a reproduction repo of your implementation?

**Comment by jacob-lester** [comment]
I still get the same error with the example project. Do you just run with `yarn start-servers` and `yarn start-gateway`

**Comment by jacob-lester** [comment]
```js
{
  "errors": [
    {
      "message": "\n      GraphQL-Modules Error: Schema is not valid!\n      - #Module #._7261207375946989 doesn't have a valid schema!\n      -- Cannot extend type \"Query\" because it is not defined.\n\nUnknown directive \"key\".\n\nUnknown type \"Query\".\n\n      Possible solutions:\n      - Check syntax errors in typeDefs\n      - Make sure you import correct dependencies\n    ",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "me"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: ",
            "      GraphQL-Modules Error: Schema is not valid!",
            "      - #Module #._7261207375946989 doesn't have a valid schema!",
            "      -- Cannot extend type \"Query\" because it is not defined.",
            "",
            "Unknown directive \"key\".",
            "",
            "Unknown type \"Query\".",
            "",
            "      Possible solutions:",
            "      - Check syntax errors in typeDefs",
            "      - Make sure you import correct dependencies",
            "    ",
            "    at GraphQLModule.get schema [as schema] (C:\\Users\\lesterjd\\Documents\\apollo-federation\\node_modules\\@graphql-modules\\core\\src\\graphql-module.ts:352:17)",
            "    at typeResolvers.(anonymous function) (C:\\Users\\lesterjd\\Documents\\apollo-federation\\node_modules\\@graphql-modules\\core\\src\\graphql-module.ts:1056:36)",
            "    at process._tickCallback (internal/process/next_tick.js:68:7)"
          ]
        }
      }
    }
  ],
  "data": {
    "me": null
  }
}
```

**Comment by ardatan** [comment]
Could you try the following versions?
```
 - @graphql-modules/core => 0.7.8-alpha-db1285d.38+db1285d
 - @graphql-modules/di => 0.7.8-alpha-db1285d.38+db1285d
```

**Comment by jacob-lester** [comment]
Updating core worked! Also, now working in my own project implementation.

**Comment by ardatan** [comment]
Fixed in 0.7.8!
